profiles/common/modules/features/nexteuropa_trackedchanges/nexteuropa_trackedchanges.admin.inc

```diff
@@ -0,0 +1,700 @@
+  // Then, do the control.
+  $operations = array(
+    array('_nexteuropa_trackedchanges_revisions_scanning_process', array('')),
+    array('_nexteuropa_trackedchanges_entity_scanning_process', array('')),
+    array('_nexteuropa_trackedchanges_table_rebuilt_process', array('')),
+  );
+
+  $batch = array(
+    'operations' => $operations,
+    'finished' => $finished_batch,
+    'title' => t('Processing disabling validation'),
+    'init_message' => t('Disabling check is starting.'),
+    'progress_message' => t('Processed @current out of @total.'),
+    'error_message' => t('Disabling check has encountered an error.'),
+  );
+  batch_set($batch);
+
+}
+
+/**
+ * Finished callbacks designed for batches.
+ *
+ * They perform fields scanning detecting tracked changing in them.
+ */
+
+/**
+ * Finished callback of the batch field for tracked changes values.
+ *
+ * It is ran in the case where change tracking buttons (CKEditore LITE) are
+ * removed from a WYSIWYG process.
+ *
+ * @see nexteuropa_trackedchanges_disable_buttons()
+ */
+function _nexteuropa_trackedchanges_remove_field_scanning_finished($success, $results, $operations) {
+  if ($success) {
+    $text_format = $results['text_format'];
+
+    // Here we do something meaningful with the results.
+    if (!empty($results['detected_entities'])) {
+      $message = t('The deactivation of the change tracking feature for the 
+        @name profile stopped because tracked changes have been detected.',
+        array('@name' => $text_format));
+      $message .= '<br />';
+      $message .= t('Please accept or reject them before proceeding to the deactivation; 
+      the list of entities with tracked changes is available !here.',
+        array('!here' => l(t('here'), 'admin/content/tracked_changes')));
+
+      drupal_set_message($message, 'error');
+    }
+    else {
+      nexteuropa_trackedchanges_disable_buttons($text_format);
+      drupal_set_message(t('Change tracking disabled on !profile WYSIWYG profile.',
+        array('!profile' => $text_format)));
+    }
+  }
+  else {
+    // An error occurred.
+    // $operations contains the operations that remained unprocessed.
+    $error_operation = reset($operations);
+    $message_args = array(
+      '%error_operation' => $error_operation[0],
+      '@arguments' => print_r($error_operation[1], TRUE),
+    );
+    $message = t('An error occurred during the deactivation process on the 
+      %error_operation with arguments: @arguments', $message_args);
+    $message .= '<br />';
+    $message .= t('The deactivation is impossible.');
+    drupal_set_message($message, 'error');
+  }
+}
+
+/**
+ * Finished callback of the batch field for tracked changes values.
+ *
+ * It is run in the case where the feature set is disabled.
+ *
+ * @see _nexteuropa_trackedchanges_disable_check_submit()
+ */
+function _nexteuropa_trackedchanges_disable_field_scanning_finished($success, $results, $operations) {
```

Comment:
For a better readability, wouldn'it be possible to reduce the number of 'else' ?
Ex: by assigning a default value to $message and use only once the function drupal_set_message() ?